online-bookstore/src/main/java/com/example/onlinebookstore/services/book/BookServiceImpl.java

```diff
@@ -125,4 +132,20 @@ private void updateBookFields(BookDTO bookDTO, BookEntity editedBook) {
         editedBook.setQrCode(bookDTO.getQrCode());
         editedBook.setInventory(bookDTO.getInventory());
     }
+
+    private GenreEntity checkIfGenreExistsInDBAndMerge(GenreEntity genreToBeSaved) {
+        List<GenreEntity> iterateRepositoryList = genreRepository.findAll();
+        boolean genreAlreadyExists = false;
+        for (GenreEntity genre : iterateRepositoryList) {
+            if (genre.getName().equals(genreToBeSaved.getName())) {
```

Comment:
Are we sure that `equals` method suits us, instead of `equalsIgnoreCase`? If we are sure that we'll always have a genre with either lower or upper case, we can go with the current approach.

Also, I didn't check on the DTO, but we are sure that we will receive a value for the genre on the API and `genreToBeSaved.getName()` will not throw an NPE?

Reply:
Now, this was one of the most challenging and confusing things, the genre is an enum class, so all the 'value' in the enum are uppercase. It's been a headache to actually get the values from the enum class and i'm still learning how to do it. The DTO receives the value but it has to be uppercase as in the enum class, get name works as i've defined a custom method for that.

Reply:
Yes, you need helper methods that work hand in hand with the enum. We can discuss around this topic on future sessions.